### PR TITLE
Documentation/282/session management

### PR DESCRIPTION
**Pull request by carltonsmith**
### This pull request changes...

This is a documentation update that details our session management strategy. This was reviewed previously on `raft-tdp-main`

### CODE REVIEW

File: tdrs-backend/docs/session-management.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+### Frontend
+The frontend will also have a timer that it will set when the user logs in. It will monitor user activity and reset every time a user interacts with the page. That means when a form field is filled out or changed, the FE timer will reset. Because this is not dependent on interactions with the BE, the FE will call the /v1/authorization-check endpoint whenever it resets the timer. This will serve to verify the user is still authenticated on the BE as well as refresh the cookie.
+
+When the FE timer reaches 20 minutes (?) it will alert the user that the session will time out soon and give them the option to either extend the session or logout of the system. If the user chooses to extend the session, the FE will call the /v1/authorization-check endpoint to refresh the session. If the user elects to log out of the system it will call the /v1/logout endpoint to clear the session and log the user out of the system. We will also employ a strategy of using `debounce` to mitigate the potential of sending too many requests to the backend.
```
Comment (alexsoble) [inline tdrs-backend/docs/session-management.md:11]: @carltonsmith I think we can remove the question mark here, `tdrs-frontend/.env.development` shows a timeout setting of 20 minutes.

File: tdrs-backend/docs/session-management.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Session Management
+
+The requirement for this project is that users will be logged out of the system after 30 minutes of activity. This is our strategy to accomplish that.
+
+### Backend
+The backend will be the ultimate arbiter of session management. When the user logs in they will receive an HttpOnly cookie that is set to expire in 30 minutes. After that, with every interaction between the FE and BE, the BE will refresh the cookie, so it will extend the timeout time to another 30 minutes.
```
Comment (alexsoble) [inline tdrs-backend/docs/session-management.md:6]: Worth mentioning how this is configured? `SESSION_TIMEOUT` variable set in `common.py`, possible to override per-env?

**Review by alexsoble — CHANGES_REQUESTED** [review]
Glad we're reviewing and merging this! Noticed one spot that has a `?`, we should clear that up before merging.

File: tdrs-backend/docs/session-management.md (hunk 1 above)
Comment (carltonsmith) [reply tdrs-backend/docs/session-management.md]: @alexsoble No problem

File: tdrs-backend/docs/session-management.md (hunk 2 above)
Comment (carltonsmith) [reply tdrs-backend/docs/session-management.md:6]: @alexsoble I have updated accordingly - good idea

**Review by alexsoble — APPROVED** [review]
# QASP Review

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

_N/A: Documentation PR._

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ [x] Both test suites pass.
+ Are code coverage minimums met?
  + [x] Frontend coverage: 100%
  + [x] Backend coverage: 100%

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Backend code style checks passing on CircleCI
+ [x] Frontend code style checks passing on CircleCI

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

_N/A: Documentation PR._

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

_N/A: Documentation PR._

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

Adds documentation of session management strategy. 

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] OWASP Scan passes on CircleCI

**Pull request merged**